server/Common/Versioning/Versioning.cs

```diff
@@ -0,0 +1,137 @@
+                {
+                    // Default to API version 1.0
+                    feature.RequestedApiVersion = v1p0;
+                }
+                else if (feature.RequestedApiVersion != v1p0)
+                {
+                    // The old Tyger CLI client does not request an API version, so we force an UnsupportedApiVersion error
+                    feature.RequestedApiVersion = new ApiVersion(0, 0);
+                }
+            }
+        }
+    }
+}
+
+/// <summary>
+/// Implements a problem details writer for emitting our own ErrorBody JSON response.
+/// Loosely based on https://github.com/dotnet/aspnet-api-versioning/blob/3fc071913dcded23eeb5ebe55bca44f3828488bf/src/AspNetCore/WebApi/src/Asp.Versioning.Http/ErrorObjectWriter.cs
+/// </summary>
+public partial class ProblemDetailsErrorBodyWriter : IProblemDetailsWriter
+{
+    private const string ProblemDetailsCodeKey = "code";
+
+    private readonly string[] _supportedApiVersions;
+
+    public ProblemDetailsErrorBodyWriter(IEnumerable<ApiVersion> supportedApiVersions)
+    {
+        ArgumentNullException.ThrowIfNull(supportedApiVersions);
+
+        _supportedApiVersions = supportedApiVersions.Select(v => v.ToString()).ToArray();
+    }
+
+    public virtual bool CanWrite(ProblemDetailsContext context)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+
+        var type = context.ProblemDetails.Type;
+
+        string? code = null;
+
+        if (type == ProblemDetailsDefaults.Unsupported.Type)
+        {
+            code = ProblemDetailsDefaults.Unsupported.Code;
+        }
+        else if (type == ProblemDetailsDefaults.Unspecified.Type)
+        {
+            code = ProblemDetailsDefaults.Unspecified.Code;
+        }
+        else if (type == ProblemDetailsDefaults.Invalid.Type)
+        {
+            code = ProblemDetailsDefaults.Invalid.Code;
+        }
+        else if (type == ProblemDetailsDefaults.Ambiguous.Type)
+        {
+            code = ProblemDetailsDefaults.Ambiguous.Code;
+        }
+
+        if (code != null)
+        {
+            // This is a workaround for Asp.Versioning BUG https://github.com/dotnet/aspnet-api-versioning/issues/1091
+            context.ProblemDetails.Extensions[ProblemDetailsCodeKey] = code;
+            return true;
+        }
+
+        return false;
+    }
+
+    public ValueTask WriteAsync(ProblemDetailsContext context)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+
+        var response = context.HttpContext.Response;
+
+        var errorCode = context.ProblemDetails.Extensions.TryGetValue(ProblemDetailsCodeKey, out var value) &&
+                   value is string code ? code : "";
+        var errorMessage = context.ProblemDetails.Title ?? "";
+
+        var obj = new ErrorBody(errorCode, errorMessage, _supportedApiVersions);
+
+        return new ValueTask(Results.Json(obj, statusCode: response.StatusCode).ExecuteAsync(context.HttpContext));
```

Comment:
More idiomatic would be to make the method async and await instead of constructing the ValueTask.